confidant/models/credential.py

```diff
@@ -142,6 +161,41 @@ def _get_decrypted_credential_pairs(self):
         _credential_pairs = json.loads(_credential_pairs)
         return _credential_pairs
 
+    @property
+    def next_rotation_date(self):
+        """
+        Return when a credential needs to be rotated for security purposes.
+        """
+        # Some credentials never need to be rotated
+        if self.exempt_from_rotation:
+            return None
+
+        # If a credential has never been rotated or been decrypted,
+        # immediately rotate
+        if self.last_rotation_date is None:
+            return datetime.utcnow()
+
+        if (self.last_decrypted_date and
+                self.last_decrypted_date > self.last_rotation_date):
+            return datetime.utcnow()
+
+        days = settings.MAXIMUM_ROTATION_DAYS
+        for tag in self.tags:
+            rotation_days = settings.ROTATION_DAYS_CONFIG.get(tag)
+            if rotation_days is None:
+                continue
+            if days is None or rotation_days < days:
+                days = rotation_days
+        return self.last_rotation_date + timedelta(days=days)
+
+    @property
+    def exempt_from_rotation(self):
+        """
+        Credentials with certain tags (say financial credential)
+        must be rotated.  Credentials without these tags can be exempt.
+        """
+        return len(set(self.tags) & set(settings.TAGS_REQUIRING_ROTATION)) == 0
```

Comment:
I much prefer this to a ROTATION_EXEMPT tag as people won't need to migrate all records - they can just tag the few they care about